### DISCUSSION TITLE
Theia and VSCode GLSP-Clients in the same monorepo don't work

### DISCUSSION THREAD

**Discussion by rhumbertgz** [eclipse-glsp/glsp#1102] [Q&A]
Hello, 

Using as started template the node-theia-json template, I added the the tasklist-vscode project. I want to have both IDE UIs in the same monorepo. Although, I don't get any compilation error, the theia-ide crashes at runtime, while the vscode-extension works perfectly. 

Here you can see the error that I get:

![Screenshot 2023-08-30 at 14 11 09](https://github.com/eclipse-glsp/glsp/assets/10393245/d070f386-1d98-413f-9850-d37e54c56916)

I have done some tests, and I found that if I remove the vscode project from my monorepo workspaces, everything works with the theia-ide. I checked the new packages installed by the vscode project and I found more than 25 differences in the versions of its dependencies in comparison with the theia projects. Sometimes the vscode project uses older deps versions and other newer versions. 

Is there known incompatibility? I've been hunting by try and error to find the conflict without success.

**Comment by rhumbertgz** [comment]
As you can see in these screenshots, I only copied the `tasklist-vscode` project from `node-json-vscode` template to `node-json-theia/glsp-client`. I also updated the `package.json` and the `tsconfig.eslint.json`

![Screenshot 2023-08-30 at 21 09 59](https://github.com/eclipse-glsp/glsp/assets/10393245/e3b3c5ef-b987-447f-9e07-9368cdb1d6ad)

![Screenshot 2023-08-30 at 21 09 22](https://github.com/eclipse-glsp/glsp/assets/10393245/49a628bb-16d5-4e95-9543-96e80898f298)

![Screenshot 2023-08-30 at 21 10 40](https://github.com/eclipse-glsp/glsp/assets/10393245/7e7826c6-a371-4ddc-90f3-fa2cc7225379)

**Reply by rhumbertgz** [reply]
Any idea what could be root of the issue?

**Reply by rhumbertgz** [reply]
After a long tracking of the issue I found the root of the problem "minimatch "^9.0.0". The`tasklist-browser-app` requires minimatch "^5.1.0" as indirect dependency of theia. However when the `tasklist-vscode ` dependencies are installed, the minimatch package inside the `tasklist-browser-app` was getting updated to "^9.0.0". So, my solution was to lock `tasklist-browser-app` by adding "minimatch "^5.1.0" to its package.json.

**Reply by rhumbertgz** [reply]
The problem is back :(

**Comment by rsoika** [comment]
In our own project we develop the main functionality in [one monorepo](https://github.com/imixs/open-bpmn) including

 - the java server part
 - the glsp client part
 - and an additional util module

This makes development of new features faster as we can focusing on one technology. 

For the vs-code integration we maintain a [separated monorepo](https://github.com/imixs/open-bpmn-vscode-integration) including

 - the vscode integration code, depending our glsp-client part and including the server jar. 

And so we do also with the [eclipse-integration](https://github.com/imixs/open-bpmn-eclipse-integration) (still in development). 

I think mixing all these parts in a big project causes more problems than it simplifies things.

**Reply by rhumbertgz** [reply]
Thanks for the recommendation @rsoika, I gave up on that idea. It is something related to webpack, when the two front-end are added  and bundled, it pushes things to the Theia-version that are not available on the browser. I tried polyfill to mitigate the issue but new ones appeared.
